vcs/none.go

```diff
@@ -0,0 +1,41 @@
+package vcs
+
+import (
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/fossas/fossa-cli/errors"
+)
+
+// A NoRepository system is used in a project that is not version controlled.
+type NoRepository struct {
+	dir        string
+	project    string
+	revisionId time.Time
+}
+
+// NewNoRepository takes the directory marking the root of a codebase that is not version controlled and
+// returns an implementation of the System interface that uses the filesystem and current time as metadata.
+func NewNoRepository(dir string) (*NoRepository, error) {
+	path, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not resolve absolute path to project root")
+	}
+	return &NoRepository{
+		dir:        dir,
+		project:    filepath.Base(path),
+		revisionId: time.Now(),
```

Comment:
I thought about this too, and I think that using the time is the most robust approach: reading and hashing an entire directory tree can be expensive and invite errors because of the huge amount of file I/O involved. In general it's not likely that anyone is often going to run fossa analyze on the same exact code more than once, especially if the CLI is plugged in to a CI environment.

Reply:
Good to hear, wanted to make sure we considered this option. It might be helpful to leave a comment in the description about this decision.